### PR TITLE
Added logger to client

### PR DESCRIPTION
**Pull request by dcolligan**
Produces these log messages:

_debugLevel 0:
nothing

_debugLevel 1:
```
ga4gh.client: POST https://www.googleapis.com/genomics/v1beta2/variants/search
```

_debugLevel 2:
```
ga4gh.client: POST https://www.googleapis.com/genomics/v1beta2/variants/search
[dump of Json response]
```

Issue #130

### CODE REVIEW

File: ga4gh/client.py [hunk 1]
```diff
@@ -28,6 +29,20 @@ def __init__(self, urlPrefix, debugLevel, workarounds, key):
         self._workarounds = workarounds
         self._key = key
 
+        # logging config
+        self.logger = logging.getLogger(__name__)
```
Comment (jeromekelleher) [inline ga4gh/client.py]: Should logger be a public attribute or protected (i.e., `self._logger`)? I'd usually default to protected.

File: ga4gh/client.py [hunk 2]
```diff
@@ -46,17 +61,14 @@ def _usingWorkaroundsFor(self, workaround):
...
     def _debugResponse(self, jsonString):
-        if self._debugLevel > 1:
-            # TODO use a logging output and integrate with HTTP client more
-            # nicely... also, logging / error handling in general
-            print("json response:")
-            pp = json.dumps(
-                json.loads(jsonString), sort_keys=True, indent=4)
-            print(pp)
+        self.logger.debug("json response:")
+        pp = json.dumps(
```
Comment (jeromekelleher) [inline ga4gh/client.py]: The trouble with this is we're doing a lot of JSON manipulation even when _debugLevel = 0, which would probably slow things down a lot. Maybe we should remove the JSON formatting? Is there a clean way to keep pretty printing (which is handy) but without having to construct the string every time?

**Comment by jeromekelleher** [comment]
Looks good, thanks @dcolligan. I've a few minor issues but I think this is generally what we want all right.

File: ga4gh/client.py (hunk 2 above)
Comment (dcolligan) [inline ga4gh/client.py]: Hm, you're right.  I would have liked to just send everything to the logger and let its internal configuration determine what and what not to emit, but it looks like we will need to add in some of our own logic for performance reasons.

@jeromekelleher do you think it makes more sense to drive the logic off of `_debugLevel` or off of logger's log levels e.g. `if self.logLevel < logger.WARN` ?  If we do the former than _debugLevel only is used in __init__ and nowhere else, which is nice... on the other hand, doing the later forces us to understand more about how logger works...

Comment (jeromekelleher) [inline ga4gh/client.py]: Your call @dcolligan --- both have benefits and drawbacks...

**Comment by dcolligan** [comment]
@jeromekelleher new commit addresses your concerns

**Comment by jeromekelleher** [comment]
Nice. +1.

**Pull request merged**